src/cpp/core/include/core/r_util/RActiveSessions.hpp

```diff
@@ -36,67 +37,101 @@ namespace r_util {
 class ActiveSession : boost::noncopyable
 {
 private:
+
    friend class ActiveSessions;
-   ActiveSession() {}
+   explicit ActiveSession(std::shared_ptr<IActiveSessionStorage> storage) {}
 
-   explicit ActiveSession(const std::string& id) : id_(id) 
+   ActiveSession(
+      std::shared_ptr<IActiveSessionStorage> storage,
+      const std::string& id) :
+         storage_(storage),
+         id_(id) 
    {
    }
 
-   explicit ActiveSession(const std::string& id, const FilePath& scratchPath)
-      : id_(id), scratchPath_(scratchPath)
+   ActiveSession(
+      std::shared_ptr<IActiveSessionStorage> storage,
+      const std::string& id,
+      const FilePath& scratchPath) : 
+         storage_(storage),
+         id_(id),
+         scratchPath_(scratchPath)
    {
       core::Error error = scratchPath_.ensureDirectory();
       if (error)
          LOG_ERROR(error);
-
-      propertiesPath_ = scratchPath_.completeChildPath("properites");
-      error = propertiesPath_.ensureDirectory();
-      if (error)
-         LOG_ERROR(error);
    }
 
-public:
+   std::string const kExecuting = "executing";
+   std::string const kInitial = "initial";
+   std::string const kLabel = "label";
+   std::string const kLastUsed = "last_used";
+   std::string const kProject = "project";
+   std::string const kSavePromptRequired = "save_prompt_required";
+   std::string const kSessionSuspendData = "suspended_session_data";
+   std::string const kRunning = "running";
+   std::string const kRVersion = "r_version";
+   std::string const kRVersionHome = "r_version_home";
+   std::string const kRVersionLabel = "r_version_label";
+   std::string const kWorkingDir = "working_directory";
+
+ public:
 
    bool empty() const { return scratchPath_.isEmpty(); }
 
    std::string id() const { return id_; }
 
    const FilePath& scratchPath() const { return scratchPath_; }
 
-   std::string project() const
+   std::string getProperty(std::string const propertyName) const
    {
+      std::string value;
       if (!empty())
-         return readProperty("project");
-      else
-         return std::string();
+      {
+         Error error = storage_->readProperty(id_, propertyName, &value);
+         if (error)
+            LOG_ERROR(error);
+      }
+
+      return value;
    }
 
-   void setProject(const std::string& project)
+   void setProperty(std::string const propertyName, std::string const value)
```

Comment:
Same as for `getProperty`
```suggestion
   void writeProperty(const std::string& propertyName, const std::string& value)
```